protostar/commands/test/testing_live_logger.py

```diff
@@ -34,21 +35,49 @@ def __init__(
         self.exit_first = exit_first
         self.slowest_tests_to_report_count = slowest_tests_to_report_count
 
-    def log_testing_summary(
-        self, test_collector_result: "TestCollector.Result"
-    ) -> None:
-        self.testing_summary.log(
-            collected_test_cases_count=test_collector_result.test_cases_count,
-            collected_test_suites_count=len(test_collector_result.test_suites),
-            slowest_test_cases_to_report_count=self.slowest_tests_to_report_count,
-        )
-
-    def log(
+    def log_json(
         self,
         shared_tests_state: SharedTestsState,
         test_collector_result: "TestCollector.Result",
+        messenger: Messenger,
     ):
+        try:
+            tests_left_n = test_collector_result.test_cases_count
+            while tests_left_n > 0:
+                test_result: TestResult = shared_tests_state.get_result()
+
+                self.testing_summary.extend([test_result])
+
+                test_result = make_path_relative_if_possible(
+                    test_result, self._project_root_path
+                )
+
+                messenger(format_test_result(test_result))
 
+                if self.exit_first and shared_tests_state.any_failed_or_broken():
+                    tests_left_n = 0
+                    return
+
+                if isinstance(test_result, BrokenTestSuiteResult):
+                    tests_in_case_count = len(test_result.test_case_names)
+                    tests_left_n -= tests_in_case_count
+                else:
+                    tests_left_n -= 1
+        finally:
+            messenger(
+                TestingSummaryResultMessage(
+                    test_collector_result=test_collector_result,
+                    testing_summary=self.testing_summary,
+                    slowest_tests_to_report_count=self.slowest_tests_to_report_count,
+                )
+            )
```

Comment:
Ideally, we shouldn't duplicate the queue consumer. This can be refactored in a different PR.

Reply:
Why did you resolve this conversation without providing a reason for it? What happened? Did you create an issue for it? Did you already post a PR? Did you change your mind about it?

Reply:
OK. #1359, resolving.